Terminal.Gui/Drawing/Color.Formatting.cs

```diff
@@ -7,418 +7,562 @@
+    ///             <description>
+    ///                 Extended general format - Returns a named <see cref="Color"/> if there is a match, or a
+    ///                 32-bit/4-byte/8-hex digit string in "#AARRGGBB" format.
+    ///             </description>
+    ///         </item>
+    ///         <item>
+    ///             <term>d</term>
+    ///             <description>
+    ///                 Decimal format - Returns a 3-component decimal representation of the <see cref="Color"/> in
+    ///                 "rgb(R,G,B)" format.
+    ///             </description>
+    ///         </item>
+    ///         <item>
+    ///             <term>D</term>
+    ///             <description>
+    ///                 Extended decimal format - Returns a 4-component decimal representation of the <see cref="Color"/> in
+    ///                 "rgba(R,G,B,A)" format.
+    ///             </description>
+    ///         </item>
+    ///     </list>
+    ///     <para>
+    ///         If <paramref name="formatProvider"/> is provided and is a non-null <see cref="ICustomColorFormatter"/>, the
+    ///         following behaviors are available, for the specified values of <paramref name="formatString"/>:
+    ///         <list type="bullet">
+    ///             <listheader>
+    ///                 <term>Value</term> <description>Result</description>
+    ///             </listheader>
+    ///             <item>
+    ///                 <term>null or empty string</term>
+    ///                 <description>
+    ///                     Calls <see cref="ICustomColorFormatter.Format(string?,byte,byte,byte,byte)"/> on the provided
+    ///                     <paramref name="formatProvider"/> with the null string, and <see cref="R"/>, <see cref="G"/>,
+    ///                     <see cref="B"/>, and <see cref="A"/> as typed arguments of type <see cref="Byte"/> .
+    ///                 </description>
+    ///             </item>
+    ///             <item>
+    ///                 <term>All other values</term>
+    ///                 <description>
+    ///                     Calls <see cref="string.Format{TArg0}"/> with the provided <paramref name="formatProvider"/> and
+    ///                     <paramref name="formatString"/> (parsed as a <see cref="CompositeFormat"/>), with the value of
+    ///                     <see cref="Argb"/> as the sole <see langword="uint"/>-typed argument.
+    ///                 </description>
+    ///             </item>
+    ///         </list>
+    ///     </para>
+    /// </remarks>
+    [SkipLocalsInit]
+    public string ToString (
+        [StringSyntax (StringSyntaxAttribute.CompositeFormat)] string? formatString,
+        IFormatProvider? formatProvider = null
+    ) {
+        return (formatString, formatProvider) switch {
+                   // Null or empty string and null formatProvider - Revert to 'g' case behavior
+                   (null or { Length: 0 }, null) => ToString (),
 
```

Comment:
Hmmm...

I am not sure if I like how the current ruleset results in these getting all these extra newlines.

I don't think it's worth bothering with trying to fix that in this PR, since it's not yet a particularly common situation, but next time I run into it I'll see if there is support for this specific situation.